### ISSUE TITLE
The ctrl files to define custom DNS server are respected only when 8.8.8.8 is not available

### ISSUE THREAD

**Issue by BrianGilbert** [omega8cc/boa#674]
For Vagrant setup it stops the guest from using my host as the upstream DNS.
It would be great if there could be a way to disable it during install/upgrades.

I essentially want to ensure that /etc/resolve.conf points to 10.0.2.2

**Comment by BrianGilbert** [comment]
So I see now that this might be caused by resolvconf, not pdnsd, I'm not sure though?

**Comment by omega8cc** [comment]
There are two ctrl files to control this: https://github.com/omega8cc/boa/blob/master/docs/ctrl/system.ctrl#L82

**Comment by BrianGilbert** [comment]
thanks

**Comment by BrianGilbert** [comment]
@omega8cc 
I've tried this and am still having some problems, here is my script that vagrant rund on provisioning of the box:

```
#!/bin/sh
# Written by Brian Gilbert @BrianGilbert_ https://github.com/BrianGilbert
# of Realityloop @Realityloop http://realitylop.com/

echo "Provisioning virtual machine..."

echo "Use host for DNS lookups..."
rm /etc/resolv.conf
echo "nameserver 10.0.2.2" > /etc/resolv.conf
cp /etc/resolv.conf /var/backups/resolv.conf.vanilla

echo "Get BOA..."
cd /root

rm BOA.sh.txt
wget -q -U iCab http://files.aegir.cc/BOA.sh.txt > /dev/null 2>&1;bash BOA.sh.txt

touch /root/.use.local.nameservers.
touch /root/.mysql.no.new.password.cnf
touch /root/.redis.no.new.password.cnf

echo "Get BOA Configurations..."
rm .o1.octopus.cnf
curl -s https://gist.githubusercontent.com/BrianGilbert/840ce65eaecae6de9be3/raw/64852dd9057043a4cff701e4c3a062bfa301ab2a/o1.octopus.cnf > .o1.octopus.cnf

rm .barracuda.cnf
curl -s https://gist.githubusercontent.com/BrianGilbert/b759280a51403d43a2ca/raw/bb4c0029d01ac1997e4f442034644195b25820a3/.barracuda.cnf > .barracuda.cnf

echo "Install BOA..."
boa in-stable public guest.vagrant.vd developers@realityloop.com o1

#echo "Ensure we are still using host for DNS lookups..."
#rm /etc/resolv.conf
#echo "nameserver 10.0.2.2" > /etc/resolv.conf

echo "Add vagarant user to users group..."
usermod -aG users vagrant
```

The octopus part of the install fails due to DNS, and /etc/resolv.conf gets populated with 127.0.0.1 even though I have performed the steps as per the system.ctrl file.

Any suggestions?

**Comment by omega8cc** [comment]
A typo? `touch /root/.use.local.nameservers.`

**Comment by BrianGilbert** [comment]
oops :/

**Comment by BrianGilbert** [comment]
Though I'm still seeing the same behaviour after updating the incorrect filename

/etc/resolv.conf still gets populated with 127.0.0.1

```
==> default: BOA [22:38:37] ==> Final post-install cleaning, one moment...
==> default: BOA [22:38:38] ==> BYE!
==> default:
==> default: BOA [22:38:46] ==> Skynet Agent v.BOA-2.4.1 welcomes you aboard!
==> default:
==> default: BOA [22:38:50] ==> New Octopus Setup on guest.vagrant.vd in progress...
==> default:
==> default:     * Your e-mail address is support@realityloop.com
==> default:     * Your client e-mail address is
==> default:     * Your Aegir control panel for this instance will be available at:
==> default:         https://o1.vagrant.vd
==> default:     * Your Aegir system user for this instance will be o1
==> default:     * This Octopus will use PHP-CLI 5.5 for all sites
==> default:     * This Octopus will use PHP-FPM 5.5 for all sites
==> default:     * This Octopus includes platforms: NONE
==> default:     * This Octopus options are listed as SSD / Y / 8 C
==> default: BOA [22:38:58] ==> INIT A: Shared platforms code v.001 (new) will be created
==> default: BOA [22:38:58] ==> INIT A: Adding user...
==> default: BOA [22:38:58] ==> INIT A: FATAL ERROR: DNS looks broken for guest.vagrant.vd
==> default: BOA [22:38:58] ==> INIT A: FATAL ERROR: Aborting AegirSetupA installer NOW!
==> default: BOA [22:38:58] ==> FATAL ERROR: AegirSetupA installer failed
==> default: BOA [22:38:58] ==> FATAL ERROR: Aborting Octopus installer NOW!
==> default: BOA in-stable completed
==> default: Bye
==> default: Add vagarant user to users group...
==> default: Configuring cache buckets...
[~/vagrant/projects/rlbox]
[brian]
10008 > vagrant ssh
Linux default 3.2.0-4-amd64 #1 SMP Debian 3.2.65-1+deb7u1 x86_64

 Barracuda System managed by Skynet Agent v.BOA-2.4.1 welcomes you aboard

You have mail.
Last login: Wed Apr  1 05:11:27 2015 from 10.0.2.2
vagrant@guest:~$ cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
nameserver 127.0.0.1
vagrant@guest:~$ sudo su
sudo: unable to resolve host guest.vagrant.vd
guest:/home/vagrant# cd /root
guest:~# ls -la
total 64K
drwx------  3 root root 4.0K Apr  4 22:38 ./
drwxr-xr-x 26 root root 4.0K Apr  4 22:21 ../
drwx------  2 root root 4.0K Apr  4 22:38 .aptitude/
-rw-r--r--  1 root root  11K Apr  4 22:04 .barracuda.cnf
-rw-------  1 root root    0 Apr  1 05:14 .bash_history
-rw-r--r--  1 root root 3.1K Apr  4 22:18 .bashrc
-rw-r--r--  1 root root 3.1K Apr  4 22:06 .bashrc.bak.150404-2204
-rw-------  1 root root   37 Apr  4 22:36 .my.cnf
-rw-------  1 root root    9 Apr  4 22:32 .my.pass.txt
-rw-r--r--  1 root root    0 Apr  4 22:04 .mysql.no.new.password.cnf
-rw-------  1 root root    9 Apr  4 22:36 .mytop
-rw-r--r--  1 root root 6.8K Apr  4 22:04 .o1.octopus.cnf
-rw-r--r--  1 root root  140 Nov 19  2007 .profile
-rw-r--r--  1 root root    0 Apr  4 22:04 .redis.no.new.password.cnf
-rw-------  1 root root    9 Apr  4 22:32 .redis.pass.txt
-rw-------  1 root root 1.0K Apr  4 22:33 .rnd
-rw-r--r--  1 root root    0 Apr  4 22:04 .use.local.nameservers.cnf
guest:~# cat /var/backups/resolv.conf.vanilla
nameserver 10.0.2.2
guest:~#

**Comment by omega8cc** [comment]
This looks like a bug then, because these control files are respected only if the system can't use Google DNS servers.

**Comment by BrianGilbert** [comment]
@omega8cc 
Are you planning to resolve this fairly soon, or hoping for a pull request?

I took a quick look but that particular issue is in several files as far as I could tell.. and I'm not 100% sure what the ideal logic would be to change it to.

**Comment by omega8cc** [comment]
It will be fixed in 2.4.2 per milestone you can see on the right.
